From 5865b17e9eed4dca901f79e71000941f28655ed0 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Thu, 3 Mar 2022 09:38:26 -0700
Subject: [PATCH 08/17] netfs: Ensure ret is always initialized in
 netfs_begin_read()

Clang warns:

  fs/netfs/io.c:630:6: error: variable 'ret' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
          if (sync) {
              ^~~~
  fs/netfs/io.c:655:9: note: uninitialized use occurs here
          return ret;
                 ^~~

There is only one instance of netfs_begin_read() where sync is false and
its return value is not checked so just return 0 in this case, as there
is no return code from another function call to propagate.

Fixes: 33e1e5d53058 ("netfs: Add a function to consolidate beginning a read")
Link: https://github.com/ClangBuiltLinux/linux/issues/1607
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20220303163826.1120936-1-nathan@kernel.org
---
 fs/netfs/io.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/netfs/io.c b/fs/netfs/io.c
index 3db9356eb7c2..4a3b7989cd1c 100644
--- a/fs/netfs/io.c
+++ b/fs/netfs/io.c
@@ -651,6 +651,7 @@ int netfs_begin_read(struct netfs_io_request *rreq, bool sync)
 		/* If we decrement nr_outstanding to 0, the ref belongs to us. */
 		if (atomic_dec_and_test(&rreq->nr_outstanding))
 			netfs_rreq_assess(rreq, false);
+		ret = 0;
 	}
 	return ret;
 }
-- 
2.35.1

